aws/src/main/java/org/apache/iceberg/aws/s3/signer/S3V4RestSignerClient.java

```diff
@@ -213,12 +214,13 @@ private AuthSession authSession() {
                       expiresAtMillis(properties()),
                       new AuthSession(
                           ImmutableMap.of(),
-                          token,
-                          null,
-                          credential(),
-                          SCOPE,
-                          oauth2ServerUri(),
-                          optionalOAuthParams())));
+                          AuthConfig.builder()
+                              .token(token)
+                              .credential(credential())
+                              .scope(SCOPE)
```

Comment:
I'd consider `SCOPE` as one of the optional params. It is optional in both [token exchange flow](https://datatracker.ietf.org/doc/html/rfc8693#name-token-exchange-request-and-) and [client credential flow](https://datatracker.ietf.org/doc/html/rfc6749#section-4.4).

Reply:
We can get the scope from the optional OAuth parameters for signer. Also a rest catalog isn't necessary with signer. But I'm OK with either one.